modules/000-common/images/check-kernel-version/src/check-kernel-version.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"log"
+	"os"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"golang.org/x/sys/unix"
+)
+
+func main() {
+	kernelConstraint := os.Getenv("KERNEL_CONSTRAINT")
+	if kernelConstraint == "" {
+		log.Fatal("ENV variable KERNEL_CONSTRAINT must be set")
+	}
+	c, err := semver.NewConstraint(kernelConstraint)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	utsname := unix.Utsname{}
+	unix.Uname(&utsname)
+	kernelVersion := string(utsname.Release[:])
+	v, err := semver.NewVersion(strings.Split(kernelVersion, "-")[0])
```

Comment:
Again, do we really need splitting here?

Reply:
yes, because this is common problem with semver - https://github.com/Masterminds/semver#working-with-prerelease-versions

Reply:
I think we need to add as a comment to your code to avoid questions in the future.